Models/TreeModel.cpp

```diff
@@ -172,6 +180,23 @@ QMimeData *TreeModel::mimeData(const QModelIndexList &indexes) const {
   return mimeData;
 }
 
+void TreeModel::insert(const QModelIndex &parent, int row, buffers::TreeNode *node) {
+  auto insertIndex = parent;
+  if (!parent.isValid()) insertIndex = QModelIndex();
+  auto *parentNode = static_cast<buffers::TreeNode *>(insertIndex.internalPointer());
+  if (!parentNode) {
+    insertIndex = QModelIndex();
+    parentNode = root;
+  }
+  beginInsertRows(insertIndex, row, row);
+  parentNode->mutable_child()->AddAllocated(node);
+  for (int j = parentNode->child_size() - 1; j > row; --j) {
```

Comment:
You could do this with a truncate and append, but that wouldn't really matter... I can't believe this is essentially the best way of inserting elements. Can you extract this logic to a template to make it easier to replace when Proto gets its shit straight?

Reply:
Alright: 37be6d6242030e7b2aa3b35d1bd5fa6c3bae5780

That's a good point, I was actually going to do this before but decided to wait. Also fundies did his own thing in `SpriteModel.cpp` because he's inserting more than just a single item. We'll address that later though because we need to come back to the sprite editor and finish up still.